test/suite/lib.rs

```diff
@@ -268,35 +283,228 @@ pub async fn icrc2_test_supported_standards(ledger: impl LedgerEnv) -> anyhow::R
 
 pub async fn icrc2_test_approve(ledger_env: impl LedgerEnv) -> anyhow::Result<Outcome> {
     let fee = transfer_fee(&ledger_env).await?;
-    let transfer_amount = Nat::from(10_000);
-    let initial_balance: Nat = transfer_amount.clone() + fee.clone() * 2;
+    let initial_balance: Nat = fee.clone() * 2;
     let p1_env = setup_test_account(&ledger_env, initial_balance.clone()).await?;
     let p2_env = ledger_env.fork();
-    let approve_amount = transfer_amount.clone() + fee.clone();
+    let p2_subaccount = Account {
+        owner: p2_env.principal(),
+        subaccount: Some([1; 32]),
+    };
+    let approve_amount = Nat::from(100);
```

Comment:
The approve amount should be relative to the balance. You never know if 100 tokens is a lot or a low amount compared to the fee/balance of the user. It might be that the fee is 1 and then this call afterwards would fail.

Reply:
I changed the approve amount to fee.